# Madagascar Bat Kobuvirus

This repo contains analyses for the Madagascar bat Kobuvirus paper. This repo is designed to document which samples from fecal metagenomes have detectable Kobuvirus. 

Sub-directories within this repo contain pipeline directions used to determine which samples are positive for Kobuviruses, as well as data used to make phylogenies. 

Pipeline for Kobuvirus [NCBI reference based detection](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/Finding-Initial-Kobuvirus-Positives.md) and [our whole genome reference based](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/BLAST_OP287812.md) detection following Kobuvirus full genome discovery. 

Within [Fig2A sub-directory](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/Fig2A/2A-Tree-Instructions.md), details on how to build our nucloetide full-genome length Picornaviridae ML tree. 

Within [Fig2B sub-directory](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/Fig2B/Fig2B-Tree-Instructions.md), details on how to build our nucleotide full-genome length Kobuvirus ML tree. 

Within [Fig2C sub-directory](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/Fig2C/Fig2C-Tree-Instructions.md), details on how to build our amino acid Open Reading Frame (ORF) Kobuvirus Tree. 

Details on how to build a Bayesian phylogenetic tree using [BEAST2](http://www.beast2.org/) linked [here](https://github.com/fgonzalez3/mada-bat-kobuvirus/blob/main/Fig3/BEAST-Tree-Instructions.md). 

Beyond the details listed above, all scripts to produce each figure should be self-contained and output results to the "final-figures" folder. Besides the above, broadly followed the [Mada-Bat-CoV pipeline](https://github.com/brooklabteam/Mada-Bat-CoV). 

